rkt/api_service.go

```diff
@@ -305,76 +325,190 @@ func getNetworks(p *pod) []*v1alpha.Network {
 	return networks
 }
 
-// getBasicPod returns *v1alpha.Pod with basic pod information.
-func getBasicPod(p *pod) *v1alpha.Pod {
+type aggregate []error
+
+// Error is part of the error interface.
+func (agg aggregate) Error() string {
+	if len(agg) == 0 {
+		// This should never happen, really.
+		return ""
+	}
+	if len(agg) == 1 {
+		return agg[0].Error()
+	}
+	result := fmt.Sprintf("[%s", agg[0].Error())
+	for i := 1; i < len(agg); i++ {
+		result += fmt.Sprintf(", %s", agg[i].Error())
+	}
+	result += "]"
+	return result
+}
+
+func fillStaticAppInfo(store *store.Store, p *pod, v1pod *v1alpha.Pod) error {
+	var errlist []error
```

Comment:
Looking more it doesn't look like a perfect fit, though you could call wrap successively (and lose the messages for all the inner errors).

I do think there's a better name than aggregate; `errorList`/`errList` seem sane.